src/ragapp/backend/models/orm/request.py

```diff
@@ -0,0 +1,18 @@
+from typing import Optional
+
+from sqlmodel import Field, Index, SQLModel
+
+
+class UserChatRequest(SQLModel, table=True):
```

Comment:
why not storing a `User` model that has a `ratelimit` object (in the ORM sense) like this:
```
"ratelimit": {
  "time_frame",
  "request_count",
}
```
I am not sure if `time_frame` is a better name for `window_frame` - where did you get that term from?

Reply:
Ah, i just inspired the name from the sliding window approach (which should have window start and stop pointer and aggregate the total request in the windows), but our one is much simpler and i think it's enough for our case. But your suggestion `time_frame` sounds easier to understand, i'll update it.
fyi, it's better to use this: https://github.com/laurents/slowapi but it's not using db or file for persistant, so i just came up with the current implementation.

Reply:
> why not storing a User model that has a ratelimit object (in the ORM sense) like this:
  "ratelimit": {
    "time_frame",
    "request_count",
  }
I'm not sure about how we are going to add more field to the User table so using a dedicated table for the request is not a bad idea. As it's frequently read and update per request so normalize it into the current table also easier to indexing and reduce the db performance when fetching the data.

Reply:
>dedicated table for the request is not a bad idea
agree

how about using two model classes then?
`User` and `UserChatRequest ` where as the later uses the `user_id` as foreign key?
`User` can then store later the usual stuff like name and email